lib/smart_proxy_dns_powerdns/dns_powerdns_main.rb

```diff
@@ -74,6 +79,18 @@ def domain_id
     end
 
     private
+    def domain
```

Comment:
Oh, hm.  You could do `klass.any_instance.expects(..)` or ignore my initialiser suggestion and instead request them both lazily - having a domain and domain_id method, which calls a third method to do the SQL query (which sets both instance variables) if it's unset.